### ISSUE TITLE
Traversal edge behavior using element embedding

### ISSUE THREAD

**Issue by curt-weber** [flutter/flutter#129664]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

1. Launch provided sample in web environment
2. Use tab key to traverse between inputs

The default [behavior](https://api.flutter.dev/flutter/widgets/TraversalEdgeBehavior.html) used on web by MaterialApp is `leaveFlutterView`.

### Expected results

Input should traverse between inputs before the flutter app, then between the inputs within the flutter app, then after the flutter app.

### Actual results

Input cycles between the flutter app and a single input following the flutter app.

### Code sample

<details><summary>Code sample</summary>

[Reproducible sample](https://github.com/curt-weber/flutter-traversal-edge-behavior)

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, 3.10.5, on Microsoft Windows [Version 10.0.19044.2965], locale en-US)
[√] Windows Version (Installed version of Windows is version 10 or higher)
[√] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
[√] Chrome - develop for the web
[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.3.5)   
[√] Android Studio (version 2021.3)
[√] VS Code (version 1.79.2)
[√] Connected device (3 available)
[√] Network resources
```

</details>

**Comment by curt-weber** [comment]
You can also witness this in the flutter [element embedding demo](https://flutter-forward-demos.web.app) by choosing the TextField screen and cycling with the tab key.

**Comment by darshankawar** [comment]
Thanks for the report @curt-weber I observed below upon following details provided.
The tab traversal goes to address bar first followed by in Flutter's textfield. 
Are you also seeing the same behavior or different than this ?

I verified on latest stable.

**Comment by curt-weber** [comment]
For me it doesn't go to the address bar, it just continues to cycle within the page but not through the expected controls. I witnessed this same behavior on stable and master.

**Comment by darshankawar** [comment]
Thanks for the update and feedback @curt-weber 
I tried with non-Flutter web app and used tab to navigate and it does go to address bar during the traversal. Can you also confirm this at your end as how the native web behavior is ? This is to narrow down the behavior is specific to Flutter or not.

Also, I see that you are on Windows and I checked on Mac, so not sure if that makes any difference or not.

**Comment by curt-weber** [comment]
I think its permissible to go to the address bar during traversal, this issue is more about cycling through the non-flutter and flutter elements. During my testing it doesn't go to the address bar though. In the embedding demo it cycles between the shadow control and the flutter textfield only.
![shadow](https://github.com/flutter/flutter/assets/116017986/e9c3491a-ce77-458b-87a1-f9e75b0ea25a)
![textfield](https://github.com/flutter/flutter/assets/116017986/6ff0376d-d11c-41d8-aa62-32f544bc5780)
If you change the Screen to `Counter` you will see that the tab key cycles differently, it goes through the controls on the left while alternating with the `Counter`s plus button.

![cycle](https://github.com/flutter/flutter/assets/116017986/c972d50c-a4be-4b6f-a0df-3be409bc9fc1)
With Input fields though the cycling is as above, it doesn't cycle through all the elements and it starts with the flutter fields.

**Comment by darshankawar** [comment]
Thanks for the update. Based on the report, keeping this issue open for team's input.

**Comment by yjbanov** [comment]
This is something we should definitely fix. We fixed this for the iframe case and for the full-page case to be able to hand focus back to the browser UI, but we haven't done this for element embedding.

**Comment by curt-weber** [comment]
Yep, saw this working for iframe (noticed while testing TextField samples in the api docs which use DartPad in an iframe). Figured I'd make the bug for element embedding since it hasn't been created yet!

**Comment by AndryHTC** [comment]
Hello everyone,

I've encountered this issue in my project as well. Has anyone come across a temporary workaround or fix for this problem while the official solution is being developed? Any assistance would be greatly appreciated.

Thank you!

**Comment by AndryHTC** [comment]
Additionally, I wanted to mention that for my use case, completely skipping the Flutter embedding is also an option for me. If there's a way to achieve this as a temporary solution, I'd be open to exploring it. 

Thanks again for any insights!

**Comment by AndryHTC** [comment]
@curt-weber any workaround for this?

**Comment by curt-weber** [comment]
Not that I'm aware of, we looked into trapping the tab key but it was fragile and too involved. This needs a proper fix from the team, it works as mentioned before with iframes so hopefully shouldn't be too difficult to address with element embedding.

**Comment by AndryHTC** [comment]
> Not that I'm aware of, we looked into trapping the tab key but it was fragile and too involved. This needs a proper fix from the team, it works as mentioned before with iframes so hopefully shouldn't be too difficult to address with element embedding.

Thank you for keeping me updated

**Comment by rtroberts** [comment]
Adding my vote to this. It has obvious accessibility implications, and the workarounds I can think of are unsatisfactory (e.g., wrapping the embedded Flutter app to hijack tabbing entirely).

**Comment by curt-weber** [comment]
The good news is I have tested this behavior on the master channel and it has improved due to the work they have been doing with `multi-view`. Hopefully the remaining issues will be addressed and it will reach stable soon.
